controllers/release/release_adapter.go

```diff
@@ -213,54 +214,59 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 		return results.ContinueProcessing()
 	}
 
-	err = a.syncResources()
+	environment, err := a.getEnvironment(releasePlanAdmission)
 	if err != nil {
 		return results.RequeueWithError(err)
 	}
 
-	binding, err := a.createOrUpdateSnapshotEnvironmentBinding(releasePlanAdmission)
+	err = a.syncResources()
```

Comment:
Resources should be sync only whenever a binding is created, so it should go inside the `if binding == nil` check.